File: src/main/java/me/glorantq/aboe/chat/server/permissions/DefaultPermissionProvider.java
package me.glorantq.aboe.chat.server.permissions;

import cpw.mods.fml.common.FMLCommonHandler;
import net.minecraft.entity.player.EntityPlayer;

public class DefaultPermissionProvider implements PermissionProvider {

    @Override
    public boolean hasPermission(EntityPlayer player, String permission) {
        return FMLCommonHandler.instance().getMinecraftServerInstance().getConfigurationManager().func_152596_g(player.getGameProfile());
    }

    @Override
    public String getGroup(EntityPlayer player) {
        return "No Group";
    }

    @Override
    public String getPrefix(EntityPlayer player) {
        return "";
    }

    @Override
    public String getSuffix(EntityPlayer player) {
        return "";
    }

    @Override
    public void registerPermission(String permission, PermissionLevel permissionLevel) {

    }
}

File: src/main/kotlin/me/glorantq/aboe/chat/server/permissions/DefaultPermissionProvider.kt
package me.glorantq.aboe.chat.server.permissions

import cpw.mods.fml.common.FMLCommonHandler
import net.minecraft.entity.player.EntityPlayer

class DefaultPermissionProvider : IPermissionProvider {

    override fun hasPermission(player: EntityPlayer, permission: String): Boolean {
        return FMLCommonHandler.instance().minecraftServerInstance.configurationManager.func_152596_g(player.gameProfile)
    }

    override fun getGroup(player: EntityPlayer): String {
        return "No Group"
    }

    override fun getPrefix(player: EntityPlayer): String {
        return ""
    }

    override fun getSuffix(player: EntityPlayer): String {
        return ""
    }

    override fun registerPermission(permission: String, permissionLevel: IPermissionProvider.PermissionLevel) {

    }

    override fun getDisplayName(player: EntityPlayer): String {
        return player.displayName
    }
}
